clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/manifest/KubernetesDeployManifestOperation.java

```diff
@@ -228,12 +228,16 @@ private List<KubernetesManifest> getManifestsFromDescription() {
       // check `manifest` for backwards compatibility until all existing stages have been updated.
       @SuppressWarnings("deprecation")
       KubernetesManifest manifest = description.getManifest();
-      log.warn(
-          "Relying on deprecated single manifest input (account: {}, kind: {}, name: {})",
-          accountName,
-          manifest.getKind(),
-          manifest.getName());
-      inputManifests = ImmutableList.of(manifest);
+
+      // manifest may be null as well, so check
+      if (manifest != null) {
```

Comment:
Hrmm, wondering if shouldn't log "Warning!  Input manifest is null and description is null.  Something is incorrect on the deployment most likely!" or similar.  I'd not have though BOTH would be null... ALSO... won't this immediately fail below on lin 242 with an NPEsince inputManifests would be null in this check?

Reply:
I forget how this all works with `description.getManifest()` vs `description.getManifests()`, but apparently inputManifests isn't null in this case, so things work.  If this did end up generating a NullPointerException, we'd have seen it.  I have a vague memory that @mattgogerly may know something about this, but I think this change is OK.  It's at least an improvement if it doesn't fix everything.

Reply:
MOSTLY calling out the next line:
```    
inputManifests = inputManifests.stream().filter(Objects::nonNull).collect(Collectors.toList());
```
inputManifests would be null, triggering an NPE if it got to the point of your null check. ....

Reply:
I guess if the `inputManifests.isEmpty()` would allow this to "work" without an NPE...  that may be how this is working...

Reply:
Yeah, I figure that's what's happening.

Reply:
*SIGH* I hate "it's working through a glitch" situations... but... approving.  I could see a "TOOD" to note to fix this... but as long as it works it works... and tests'll catch it hopefully!